### PR TITLE
update generic transformer to agreed upon schema

### PR DESCRIPTION
**Pull request by l-moamen**
Schema has been approved, applying necessary changes

### CODE REVIEW

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py [hunk 1]
```diff
@@ -13,24 +15,24 @@ def __init__(self,
...
+                 module_name: str):
         super(GenericRecommendationFormatterTransformer, self).__init__()
         self.group_key = group_key
         self.recommendations_col = recommendations_col
-        self.item_type = item_type
+        self.item_type = item_type.upper()
         self.module_name = module_name
-        self.module_version = module_version
+        assert self.item_type in [x.upper() for x in [c.ALBUM, c.ARTIST, c.PLAYLIST, c.TRACK, c.MIX, c.RADIO,
+                                                      "ARTIST_TOP_TRACKS", "MASTER_BUNDLE_ID", "TRACK_GROUP"]]
```
Comment (l-moamen) [inline tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:25]: using upper case with under score for the supported types

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py [hunk 2]
```diff
@@ -13,24 +15,24 @@ def __init__(self,
...
 
     def _transform(self, dataset: DataFrame) -> DataFrame:
         return (dataset
                 .withColumn(c.TYPE, F.lit(self.item_type))
-                .withColumn(c.MODULE_ID, F.lit(f"{self.module_name}_v{self.module_version}"))
-                .select(F.col(self.group_key).alias(c.HASH_KEY), c.TYPE, c.MODULE_ID,
+                .withColumn(c.MODULE_ID, F.lit(self.module_name))
+                .select(F.col(self.group_key).cast(T.StringType()).alias(c.HASH_KEY), c.TYPE, c.MODULE_ID,
```
Comment (l-moamen) [inline tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:31]: all ids should be string

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py [hunk 3]
```diff
@@ -13,24 +15,24 @@ def __init__(self,
...
+        self.item_type = item_type.upper()
         self.module_name = module_name
-        self.module_version = module_version
+        assert self.item_type in [x.upper() for x in [c.ALBUM, c.ARTIST, c.PLAYLIST, c.TRACK, c.MIX, c.RADIO,
+                                                      "ARTIST_TOP_TRACKS", "MASTER_BUNDLE_ID", "TRACK_GROUP"]]
 
     def _transform(self, dataset: DataFrame) -> DataFrame:
         return (dataset
                 .withColumn(c.TYPE, F.lit(self.item_type))
-                .withColumn(c.MODULE_ID, F.lit(f"{self.module_name}_v{self.module_version}"))
```
Comment (l-moamen) [inline tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:28]: removed versioning

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py (hunk 1 above)
Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:25]: NIT: Should: `"ARTIST_TOP_TRACKS", "MASTER_BUNDLE_ID", "TRACK_GROUP"` be in constants here?

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py (hunk 3 above)
Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:28]: Good, I'll remove versioning from my pipeline

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py (hunk 2 above)
Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:31]: Thanks, I was casting to string on my side of the pipeline, I'll remove it :)

File: tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py (hunk 1 above)
Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter_transformer.py:25]: Will `"MASTER_BUNDLE_ID"` be a valid type for the BE ?

**Review by sqafabiano — COMMENTED** [review]
Looks good, just want to have confirmation about the types we can send to BE

**Comment by l-moamen** [comment]
> Will "MASTER_BUNDLE_ID" be a valid type for the BE ?
yup we agreed in that 
> NIT: Should: "ARTIST_TOP_TRACKS", "MASTER_BUNDLE_ID", "TRACK_GROUP" be in constants here?
yahhh I looked at this file and when we have multi word constants we use camel case, so left them as strings, but I am open as Celeste metadata doesn't seem to follow the rule  
> Thanks, I was casting to string on my side of the pipeline, I'll remove it :) 
same, will remove it as well

**Review by sqafabiano — APPROVED** [review]
Let's go!

**Pull request merged**